core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,7 +64,23 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // Evaluate partitions in parallel. Partitions of each rdd will be cached by the `partitions`
+  // val in `RDD`.
+  private[spark] lazy val parallelPartitionEval: Boolean = {
```

Comment:
I'm also curious to use the default thread pool, and I like the idea not adding extra knobs to tune. However, in the case, should we still have a hard coded default threshold since when the # of rdds are small, we don't want the overhead. 

If we just make `rdds.zipWithIndex` parallel, it will be hard to test if this path is touched. What do you think?

Thanks.